service-registry/src/main/java/io/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java

```diff
@@ -122,6 +123,36 @@ private boolean retry(RequestContext requestContext, Handler<RestResponse> respo
         };
     }
 
+    static class ResponseWrapper {
+        HttpClientResponse response;
+
+        Buffer bodyBuffer;
+    }
+
+    // temporary copy from syncHandler
+    // we will use swagger invocation to replace RestUtils later.
+    private Handler<RestResponse> syncHandlerEx(CountDownLatch countDownLatch, Holder<ResponseWrapper> holder) {
+        return restResponse -> {
+            RequestContext requestContext = restResponse.getRequestContext();
+            HttpClientResponse response = restResponse.getResponse();
+            if (response == null) {
+                // invoke failed, call another SC instance
+                if (!retry(requestContext, syncHandlerEx(countDownLatch, holder))) {
```

Comment:
How about the service center doesn't send the response back. 
Do we have timeout mechanism for CountDownLatch?
BTW, is it enough if we just retry once?